pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -185,6 +185,8 @@ func (c *Reconciler) reconcile(ctx context.Context, tr *v1alpha1.TaskRun) error
 				Message: fmt.Sprintf("References a Task %s that doesn't exist: %v",
 					fmt.Sprintf("%s/%s", tr.Namespace, tr.Spec.TaskRef.Name), err),
 			})
+			c.Recorder.Eventf(tr, corev1.EventTypeWarning, "BuildCreationFailed", "Failed to create build %q: %v", tr.Name, err)
+			c.Logger.Errorf("Failed to create build for task %q :%v", err, tr.Name)
```

Comment:
oooo do you think with this change we could remove the parts of our tests that assert on logs? e.g.

https://github.com/knative/build-pipeline/blob/master/pkg/reconciler/v1alpha1/pipelinerun/pipelinerun_test.go#L277

we could assert on events instead, and completely remove the log watching!

(feel free to ignore the scope creep)

Reply:
This needs refactor in building controller to inject fakerecorder. Can I handle this in another PR?

Reply:
it okay to be another PR, but can you create an issue for it please

Reply:
https://github.com/knative/build-pipeline/issues/186